pkg/render/dex_config.go

```diff
@@ -234,8 +243,12 @@ func (d *dexBaseCfg) RequestedScopes() []string {
 }
 
 func (d *dexBaseCfg) RequiredSecrets(namespace string) []*corev1.Secret {
-	secrets := []*corev1.Secret{
-		secret.CopyToNamespace(namespace, d.tlsSecret)[0],
+	var secrets []*corev1.Secret
+	if d.tlsSecret != nil {
```

Comment:
Before these changes, whether the rendered component was Dex itself, or simply a server that should trust dex, it would always mount the secret that contained the private key. Now since these changes, Dex will mount the tls secret, and the other components will ount the certSecret in stead.